pkg/apis/servicecatalog/validation/serviceclass.go

```diff
@@ -26,7 +26,7 @@ import (
 	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
-const serviceClassNameFmt string = `[-a-zA-Z0-9]+`
+const serviceClassNameFmt string = `[-.a-zA-Z0-9]+`
```

Comment:
This change is compatible with Kubernetes names, see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
> By convention, the names of Kubernetes resources should be up to maximum length of 253 characters and consist of lower case alphanumeric characters, -, and ., but certain resources have more specific restrictions.

Even though it's not critical now, it might be important in the future.

Reply:
@nilebox Thanks for the review! 

This change brings service catalog up-to-speed on the OSB PR#452 to support period ('.') characters in service brokers' service class/plan names.

This is helps in integrating some existing service brokers which use the '.' characters in the names without having to the changes the brokers' catalog (which might cause some migration problems).

Reply:
@amshuman-kr yeah it's all good, I just left the comment for other reviewers :)

Reply:
do we need to escape the `.` so its not "any char" ?

Reply:
`.` is considered a normal character inside regex character classes. So, no escaping is required.

Just to be sure, I have added a test case where `_` (underscore) is detected as an invalid character.

Reply:
great thanks